### ISSUE TITLE
Missing released BDD-X PGM checkpoint (pgm/ckpts/pgm/bddx_weights.npy)

### ISSUE THREAD

**Issue by jzstark** [AI-secure/SafeAuto#6]
Hello SafeAuto authors,

I am reproducing the released BDD-X evaluation:
    `bash scripts/eval_bddx.sh 0.01 0.35 true 2`

The evaluator requires `pgm/ckpts/pgm/bddx_weights.npy`. However, this checkpoint is not present in the GitHub repository or the released Hugging Face SafeAuto-BDDX model files.

The released pgm/predicates/bddx/train_vectors.pkl is 36-dimensional, while the BDDX MLN rules index the 52-dimensional predicate space, so `python -m pgm.pgm --dataset bddx` fails with an IndexError. Could you please release the BDD-X PGM checkpoint, or provide:

  1. a download URL for bddx_weights.npy,
  2. its SHA-256 checksum,
  3. the exact commit/dataset revision and training command used to produce it,
  4. clarification on how the 52-dimensional training predicates were generated?

  Thank you!

**Comment by javyduck** [comment]
Hi, thanks for the interest in our work and for the very careful digging — sorry for the missing checkpoint! Everything should be resolved on the latest `main`:

1. **Checkpoint released**: the original BDD-X PGM checkpoint used in the paper is now included in the repo at `pgm/ckpts/pgm/bddx_weights.npy` (sha256: `f142fae50c0774c26985f6860eb8c0a77f558b25130418b6e06d449eaed9ed86`), so `bash scripts/eval_bddx.sh 0.01 0.35 true 2` works out of the box.
2. **36-dim vs 52-dim**: the released `train_vectors.pkl` contains only the *observed* predicates (16 actions + 12 environment + 8 control signals = 36 dims). The original checkpoint was trained with the 16 MLLM action predicates additionally included — they are generated from the MLLM's predicted actions on the training set, giving the full 52-dim space.
3. **Retraining with the released vectors**: `python -m pgm.pgm --dataset bddx` now works — `pgm/pgm.py` automatically zero-pads the missing MLLM block (those rules are trivially satisfied during training and only take effect at inference), so the `IndexError` is gone. The training loop is also vectorized now and finishes in seconds. We additionally fixed a `PGM(weights_path=...)` typo in `eval_custom_predsig_bddx.py`.

Please pull the latest `main` and let me know if anything else comes up. Thanks again for the detailed report!
